### ISSUE TITLE
`osc project` doesn't check context first

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#1349]
If I have a context set called "foo" and I do `osc project foo`, I should go into that context.  The current code does not do that - I'm pretty sure I added that as a comment on that pull request.

**Comment by fabianofranz** [comment]
@smarterclayton When doing `osc project <something>` try to use a context first, correct? In case you have a ctx *and* a project with the same name.

**Comment by smarterclayton** [comment]
Precedence (i think):

User says `osc project foo`

1. I have a context called "foo" -> switch to it, no calls to the master
2. Check for a project called "foo" on the current server and user
   1. If one exists, create a new context "foo" pointing to project foo, reuse current server and user
   2. If none exist, tell the user (and list the contexts you have)

When you switch projects, display the context name AND project name if they differ.  Also list the server and user.

If you pass osc projects -a or --all, list all contexts (do the same display thing as above).

Basically, "projects" switches contexts based on the name of projects.

----- Original Message -----

**Comment by fabianofranz** [comment]
@smarterclayton Sounds good, pretty much what I was thinking about precedence. Let me check how subsequent commands will behave when switching to a ctx without calling master and that context behaves like session expired, context does not make a namespace explicit, etc. But should be good.

**Comment by fabianofranz** [comment]
@smarterclayton BTW I'm finding a little confusing how the naming logic works when saving to config files - clusters named `localhost:8443` and contexts named `localhost:8443-ffranz`. I think I'll change it to replace `:` and `.` and make it `name: localhost-8443` for example. Is it just me?

**Comment by smarterclayton** [comment]
I do like that better.  Maybe the first server should have no port, but later ones should have (choose context name in order of ease of use, with more specificity if someone claimed the general name).

**Comment by smarterclayton** [comment]
We need to have stuff upstream for error handling, I want to see that happen independently.

**Comment by fabianofranz** [comment]
PR opened: https://github.com/openshift/origin/pull/1405. See comments.
Not yet addressing the naming logic change.
